cli/CommandLineParser.php

```diff
@@ -0,0 +1,277 @@
+						if (filter_var($value, FILTER_VALIDATE_BOOL, FILTER_NULL_ON_FAILURE) === null) {
+							$output->errors[$name] = 'invalid input: ' . $input['aliasUsed'] . ' must be a boolean';
+						}
+						break;
+				}
+			}
+		}
+
+		return $output;
+	}
+
+	/**
+	 * @template T
+	 * @param T $output
+	 * @return T
+	 */
+	private function appendTypedValidValues($output) {
+		foreach ($this->inputs as $name => $input) {
+			$values = $input['values'] ?? $input['defaultInput'] ?? null;
+			$types = $this->options[$name]->getTypes();
+			if ($values) {
+				$validValues = [];
+				$typedValues = [];
+
+				switch ($types['type']) {
+					case 'string':
+						$typedValues = $values;
+						break;
+					case 'int':
+						$validValues = array_filter($values, static fn($value) => ctype_digit($value));
+						$typedValues = array_map(static fn($value) => (int) $value, $validValues);
+						break;
+					case 'bool':
+						$validValues = array_filter($values, static fn($value) => filter_var($value, FILTER_VALIDATE_BOOL, FILTER_NULL_ON_FAILURE) !== null);
+						$typedValues = array_map(static fn($value) => (bool) filter_var($value, FILTER_VALIDATE_BOOL, FILTER_NULL_ON_FAILURE), $validValues);
+						break;
+				}
+
+				/** @var stdClass $output */
+				$output->$name = $types['isArray'] ? $typedValues : array_pop($typedValues);
+			}
+		}
+
+		/** @var T $output */
+		return $output;
+	}
+
+	/** @param array<string,string|false>|false $getoptOutput */
+	private function getoptOutputTransformer(false|array $getoptOutput): void {
+		$getoptOutput = is_array($getoptOutput) ? $getoptOutput : [];
+
+		foreach ($getoptOutput as $alias => $value) {
+			foreach ($this->options as $name => $data) {
+				if (in_array($alias, $data->getAliases(), true)) {
+					$this->inputs[$name]['aliasUsed'] = $alias;
+					$this->inputs[$name]['values'] = $value === false
+						? [$data->getOptionalValueDefault()]
+						: (is_array($value)
+							? $value
+							: [$value]);
+				}
+			}
+		}
+	}
+
+	/**
+	 * @param array<string> $userInputs
+	 * @return array<string>
+	 */
+	private function getAliasesUsed(array $userInputs, string $regex): array {
+		$foundAliases = [];
+
+		foreach ($userInputs as $input) {
+			preg_match($regex, $input, $matches);
+
+			if(!empty($matches['short'])) {
+				$foundAliases = array_merge($foundAliases, str_split($matches['short']));
+			}
+			if(!empty($matches['long'])) {
+				$foundAliases[] = $matches['long'];
+			}
+		}
+
+		return $foundAliases;
+	}
+
+	/**
+	 * @template T
+	 * @param array<string> $input List of user command-line inputs.
+	 * @param T $output
+	 * @return T
+	 */
+	private function appendUnknownAliases(array $input, $output) {
+		$valid = [];
+		foreach ($this->options as $option) {
+			$valid = array_merge($valid, $option->getAliases());
+		}
+
+		$sanitizeInput = $this->getAliasesUsed($input, $this->makeInputRegex());
+		$unknownAliases = array_diff($sanitizeInput, $valid);
+		if (empty($unknownAliases)) {
```

Comment:
```suggestion
		if ($unknownAliases === []) {
```

Reply:
I can also better like the original, which is more robust

Reply:
see https://github.com/FreshRSS/FreshRSS/pull/6099#discussion_r1495970870